test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/RepositorySignatureInfoUtilityTests.cs

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using NuGet.Common;
+using NuGet.Packaging.Core;
+using NuGet.Packaging.Signing;
+using Test.Utility.Signing;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    public class RepositorySignatureInfoUtilityTests
+    {
+        private static SignedPackageVerifierSettings _defaultSettings = SignedPackageVerifierSettings.GetDefault();
+        private static SignedPackageVerifierSettings _verifyCommandDefaultSettings = SignedPackageVerifierSettings.GetVerifyCommandDefaultPolicy();
+
+        [Fact]
+        public void GetSignedPackageVerifierSettings_NullCommonSettingsThrows()
+        {
+            // Arrange & Act
+            Action action = () => RepositorySignatureInfoUtility.GetSignedPackageVerifierSettings(null, null);
+
+            // Assert
+            action.ShouldThrow<ArgumentNullException>();
+        }
+
+        [Fact]
+        public void GetSignedPackageVerifierSettings_NullRepoSignatureInfoReturnsCommonSettings()
+        {
+            // Arrange
+            var commonSettings = _defaultSettings;
+
+            // Act
+            var settings = RepositorySignatureInfoUtility.GetSignedPackageVerifierSettings(null, commonSettings);
+
+            // Assert
+            settings.Should().Be(commonSettings);
+        }
+
+        [Fact]
+        public void GetSignedPackageVerifierSettings_RepoSignatureInfoTrueAllSignedClearsAllowUnsignedIfSet()
```

Comment:
I recommend that you follow this test naming pattern:

    <member under test>_<test condition>_<expected result>

This:

    GetSignedPackageVerifierSettings_IfAllRepositorySignedTrue_SetsAllowUnsignedFalse

...is more readable.